In [18]:
import math

a,b,c,d,e,f,g=[],[],[],[],[],[],[]

a=[[4,7],[1,2],[5,6]]
b=[[4,3,7],[1,2,7],[1,0,4]]
c=[[3,6,1]]
d=[[9,4,3,-6],[2,-1,7,5]]
e=[[1,5,8],[7,2,3],[4,0,6]]
f=[[3,0,1],[1,7,3]]
g=[[7,6,4]]

def get_matrizIdentidad(x):
    
    id=inicilizar_Matriz(x,x)
    i,j=0,0
    
    while i<x:
        while j<x:
            if(i==j):
                id[i][j]=1
            else:
                id[i][j]=0
                
            j+=1
        j=0    
        i+=1

    return id

def inicilizar_Matriz(fil,col):
    
  
    mat= [None] * fil
    
    if(fil>0):
        for i in range(fil):
            mat[i] = [None] * col
            
    return mat

    
def imprimir_matriz(a):
    
    j=0
    
    while j<len(a):
        print(a[j])
        j+=1

def suma_matrices(a,b):
    
    c=[]
    i,j=0,0
    
    if (len(a)==len(b)) and (len(a[0])==len(b[0])): 
        
        c=inicilizar_Matriz(len(a),len(a[0]))
        
        while i<len(a):
    
            while j<len(a[0]):
                c[i][j]=a[i][j]+b[i][j]
               
                j+=1
            j=0
            i+=1
    
    else:
        print("matrices de dimensiones diferentes")

    return c

def resta_matrices(a,b):
    
    c=[]
    i,j=0,0
    
    if (len(a)==len(b)) and (len(a[0])==len(b[0])): 
        
        c=inicilizar_Matriz(len(a),len(a[0]))
        
        while i<len(a):
    
            while j<len(a[0]):
                c[i][j]=a[i][j]-b[i][j]
               
                j+=1
            j=0
            i+=1
            
    else:
        print("a y b tienen diferentes dimensiones")

    return c

def multiplicacion_escalar_matriz(a,esc):
    
    i,j=0,0
     
    c=inicilizar_Matriz(len(a),len(a[0]))

    while i<len(a):
        while j<len(a):
            c[i][j]=a[i][j]*esc
            j+=1
        j=0
        i+=1
    
    return c

def transpuesta(a):
    
    c=inicilizar_Matriz(len(a[0]),len(a))

    i,j=0,0
    
    while i<len(a):
        while j<len(a[i]):
            c[j][i]=a[i][j]
            j+=1
        j=0
        i+=1

        
    return c
    
def multiplicacion_matrices(a,b):
    
    c=[]
    
    fil_a,col_a=len(a),len(a[0])
    fil_b,col_b=len(b),len(b[0])
    
    i,j,w,x,y,aux=0,0,0,0,0,0
    
    #print(fil_a,col_a,fil_b,col_b)

    if(col_a==fil_b):
        
        c=inicilizar_Matriz(fil_a,col_b)
            
        if(fil_a>col_b):
            x=fil_a
        else:
            x=col_b
        
        while i<x:
            
            while w<col_b:
                while j<fil_b:
                    aux+=a[y][j]*b[j][w]
                    j+=1
                j=0
                   
                c[y][w]=aux
             
                aux=0
                w+=1
            w=0
            if(y<fil_a):
                y+=1
            
            i+=1
    else:
        print("No es posible multiplicar las dos matrices")
    
    return c

                                    
                                    
print("1)")
imprimir_matriz(suma_matrices(e,b))
print("2)")
imprimir_matriz(suma_matrices(a,f))
print("3)")
imprimir_matriz(resta_matrices(b,e))
print("4)")
imprimir_matriz(multiplicacion_escalar_matriz(b,7))
print("5)")
imprimir_matriz(multiplicacion_matrices(e,b))
print("6)")
imprimir_matriz(transpuesta(c))
print("7)")
imprimir_matriz(multiplicacion_matrices(b,a))
print("8)")
imprimir_matriz(transpuesta(d))
print("9)")
imprimir_matriz(multiplicacion_matrices(a,c))
print("10)")
imprimir_matriz(multiplicacion_matrices(get_matrizIdentidad(len(b)),b))
print("11)")
imprimir_matriz(multiplicacion_matrices(transpuesta(e),e))
print("12)")
imprimir_matriz(multiplicacion_matrices(transpuesta(c),c))


1)
[5, 8, 15]
[8, 4, 10]
[5, 0, 10]
2)
matrices de dimensiones diferentes
3)
[3, -2, -1]
[-6, 0, 4]
[-3, 0, -2]
4)
[28, 21, 49]
[7, 14, 49]
[7, 0, 28]
5)
[17, 13, 74]
[33, 25, 75]
[22, 12, 52]
6)
[3]
[6]
[1]
7)
[54, 76]
[41, 53]
[24, 31]
8)
[9, 2]
[4, -1]
[3, 7]
[-6, 5]
9)
No es posible multiplicar las dos matrices
10)
[4, 3, 7]
[1, 2, 7]
[1, 0, 4]
11)
[66, 19, 53]
[19, 29, 46]
[53, 46, 109]
12)
[9, 18, 3]
[18, 36, 6]
[3, 6, 1]


9.2

a) ¿Cuáles son las dimensiones de las matrices?
b) Identifique las matrices cuadrada, columna y renglón.
c) ¿Cuáles son los valores de los elementos a12, b23, d32, e22,
f12 y g12?
d) Ejecute las operaciones siguientes:

1) [E] + [B] 7) [B] × [A]
2) [A] + [F] 8) [D]T
3) [B] – [E] 9) [A] × {C}
4) 7 × [B] 10) [I] × [B]
5) [E] × [B] 11) [E]T [E]
6) {C}T 12) {C}T {C}

Solucion

a)
A-->3x2 
B-->3x3  
c-->3x1
D-->2x4
E-->3x3
F-->2x3
G-->1x3

b) 
B,E --->matrices cuadradas
C --->matriz columna
G --->matriz renglon

c)
a12=7
b23=7
d32-->la matriz D solo cuenta con dos filas
e22=2
f12=0
g12=6

d)
1)
[5, 8, 15]
[8, 4, 10]
[5, 0, 10]
2)
matrices de dimensiones diferentes
3)
[3, -2, -1]
[-6, 0, 4]
[-3, 0, -2]
4)
[28, 21, 49]
[7, 14, 49]
[7, 0, 28]
5)
[17, 13, 74]
[33, 25, 75]
[22, 12, 52]
6)
[3]
[6]
[1]
7)
[54, 76]
[41, 53]
[24, 31]
8)
[9, 2]
[4, -1]
[3, 7]
[-6, 5]
9)
No es posible multiplicar las dos matrices
10)
[4, 3, 7]
[1, 2, 7]
[1, 0, 4]
11)
[66, 19, 53]
[19, 29, 46]
[53, 46, 109]
12)
[9, 18, 3]
[18, 36, 6]
[3, 6, 1]